source/jormungandr/jormungandr/instance.py

```diff
@@ -156,6 +158,18 @@ def __init__(
 
         self.zmq_socket_type = zmq_socket_type
 
+        self.equipment_providers = {}
+
+        self.configure_equipment_details_providers(instance_equipment_providers)
+
+    def configure_equipment_details_providers(self, instance_equipment_providers):
+        for provider_key in instance_equipment_providers:
+            provider = equipment_provider_manager.get_providers(provider_key)
+            if provider:
+                self.equipment_providers[provider_key] = provider
```

Comment:
This won't work once the configuration will be stored in the database.
In the instance you can just keep the keys of the wanted provider (later we will get them from the database) and get the provider from the manager each time we need them.

In the instance we will keep the list of provider id wanted, and a ref to the manager, then when needed we will get the provider from the manager, this way we always get an up to date provider without having to handle complex update system inside each instance.

Reply:
:heavy_check_mark: